brave/src/main/java/com/linecorp/armeria/common/brave/RequestContextCurrentTraceContext.java

```diff
@@ -248,15 +221,29 @@ public String toString() {
      * Armeria code should always have a request context available, and this won't work without it.
      */
     @Nullable
-    private static RequestContext getRequestContextOrWarnOnce() {
+    private RequestContext getRequestContextOrWarnOnce() {
+        if (Boolean.TRUE.equals(THREAD_NOT_REQUEST_THREAD.get())) {
+            return null;
+        }
+        if (!nonRequestThreadPatterns.isEmpty()) {
+            final String threadName = Thread.currentThread().getName();
+            for (Pattern pattern : nonRequestThreadPatterns) {
+                if (pattern.matcher(threadName).lookingAt()) {
```

Comment:
How about using `find()` instead of `lookingAt()`? I guess a user would expect `"[Rr][Mm][Ii]"` to match the threads whose names contain `RMI` at any position.

Reply:
I think `find` is too lenient - it would also match `armiria-worker` ;) And I don't think there's a way to opt-out of such lenient matching when we use `find`, even `^[Rr][Mm][Ii]` would return `true` for `find`. On the flip side, right now we can opt-in to lenient matching by prefixing with `.*`.

Reply:
Hmm? ^rmi does not match armeria

Reply:
Ah it doesn't - I don't understand `find`'s javadoc then 😅